cfme/provisioning.py

```diff
@@ -255,24 +255,25 @@ def do_vm_provisioning(template_name, provider, vm_name, provisioning_data, requ
         {version.LOWEST: 'VM Provisioned Successfully',
          "5.3": 'Vm Provisioned Successfully', })
 
-    # Wait for e-mails to appear
-    def verify():
-        if current_version() >= "5.4":
-            approval = dict(subject_like="%%Your Virtual Machine configuration was Approved%%")
-        else:
-            approval = dict(text_like="%%Your Virtual Machine Request was approved%%")
-        return (
-            len(
-                smtp_test.get_emails(**approval)
-            ) > 0
-            and len(
-                smtp_test.get_emails(
-                    subject_like="Your virtual machine request has Completed - VM:%%%s" % vm_name
-                )
-            ) > 0
-        )
-
-    wait_for(verify, message="email receive check", delay=5)
+    if smtp_test:
+        # Wait for e-mails to appear
+        def verify():
+            if current_version() >= "5.4":
+                approval = dict(subject_like="%%Your Virtual Machine configuration was Approved%%")
+            else:
+                approval = dict(text_like="%%Your Virtual Machine Request was approved%%")
+            return (
+                len(
+                    smtp_test.get_emails(**approval)
+                ) > 0
+                and len(
+                    smtp_test.get_emails(
```

Comment:
@mfalesni what do you think to moving smtp testing out into a generic function that soft asserts?